libdevcore/Algorithms.h

```diff
@@ -75,4 +75,28 @@ class CycleDetector
 	V const* m_firstCycleVertex = nullptr;
 };
 
+template<typename V>
+struct BreadthFirstSearch
+{
+	template<typename ForEachChild>
+	BreadthFirstSearch& run(ForEachChild&& _forEachChild)
+	{
+		while(!verticesToTraverse.empty())
+		{
+			V const* v = *verticesToTraverse.begin();
+			verticesToTraverse.erase(verticesToTraverse.begin());
+			visited.insert(v);
+
+			_forEachChild(*v, [this](V const& _vertex) {
+				if (!visited.count(&_vertex))
+					verticesToTraverse.insert(&_vertex);
+			});
+		}
+		return *this;
+	}
+
+	std::set<V const*> verticesToTraverse;
+	std::set<V const*> visited{};
```

Comment:
Since I was nagging in another PR about this: the brackets here are in fact completely superfluous, but our compiler settings will issue a warning when using aggregate initialization with only one element, so for this reason I nevertheless used them here :-).

Reply:
Perhaps add a constructor instead?

Reply:
Ah, well... if I add a constructor I cannot use aggregate initialization anymore and to me it seems nice to be able to do something like:
``BreadthFirstSearch<Node>{{root},{excluded_vertex1, excluded_vertex2}}.run(...)``
(and even with zero overhead)

Reply:
Ok!